File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int nextCover = 0;
        int currCover = 0;
        int count = 0;
        for (int i = 0; i < nums.size() - 1; i++) {
            nextCover = max(nextCover, nums[i] + i);
            if (i == currCover) {
                currCover = nextCover;
                count++;
                if (currCover > nums.size() - 1) {
                    return count;
                }
            }
        }

        return count;
        
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return 0

        count = 1
        currReach = nums[0]
        currStart = 0

        while currReach < n - 1:
            nextReach = 0

            for i in range(currStart, currReach + 1):
                nextReach = max(nextReach, i + nums[i])

            count += 1
            currStart = currReach + 1
            currReach = nextReach

        return count